dev/libsqlite3-pcre/build.sh

```diff
@@ -0,0 +1,87 @@
+#!/usr/bin/env bash
+
+## This script will ensure that the libsqlite3-pcre dynamic library exists in the root of this
+## repository (either libsqlite3-pcre.dylib for Darwin or libsqlite3-pcre.so for linux). This
+## script is used by run the symbol service locally, which compiles against the shared library.
+##
+## Invocation:
+## - `./libsqlite3-pcre/build.sh`         : build the library
+## - `./libsqlite3-pcre/build.sh libpath` : output its path
+
+cd "$(dirname "${BASH_SOURCE[0]}")/../.."
+set -eu
+
+OUTPUT=`mktemp -d -t sgdockerbuild_XXXXXXX`
+cleanup() {
+    rm -rf "$OUTPUT"
+}
+trap cleanup EXIT
+
+function libpath() {
+    case "$OSTYPE" in
+        darwin*)
+            echo "$PWD/libsqlite3-pcre.dylib"
+            ;;
+
+        linux*)
+            echo "$PWD/libsqlite3-pcre.so"
+            ;;
+
+        *)
+            echo "Unknown platform $OSTYPE"
+            exit 1
+            ;;
+    esac
+}
+
+function build() {
+    libsqlite3PcrePath=$(libpath)
```

Comment:
This is pre-existing in a bunch of these scripts, but these Pascal-case names are unusual for shell scripts. That aside, though, the larger concern I have is that these are defining globals inside a local scope. That's clearly not intended, since the plumbing is all there to compute the value by platform. I suggest:

```suggestion
    local sqlite_path=$(libpath)
```

(even if you leave the name the same, the `local` bit is the point I care about)

Reply:
It's also slightly amusing to me that the local variable is longer than the function it calls. 😀